### DISCUSSION TITLE
Question About RGB and sRGB with PNG

### DISCUSSION THREAD

**Discussion by hollinwilkins** [nothings/stb#1661] [Q&A] [answered]
I am not an expert in gamma correction, it seems like a topic with a lot of history. I am using STBI to write PNG files. I generate RGB values in linear space, perform gamma correction using my own code, then write the 3 u8 components into the buffer. So the STBI image data buffer holds sRGB values, NOT linear values. I then write the image using stbi_write_png. Is this the correct way to handle gamma correction when writing PNGs?

**Comment by nothings** [comment, marked as answer]
It's the appropriate solution when using stb_image_write.

However, be advised that the PNG spec does support metadata that explicitly indicates a PNG is sRGB, but stb_image_write does ~~not~~ support writing that metadata. Nevertheless, with most PNG readers this should be fine as they'll default to assuming the PNG is sRGB.
